receive.c

```diff
@@ -391,13 +436,14 @@ int main(int argc, char **argv) {
 		}
 	}
 
-	if(filteropt == 1) {
+	if(filtertype > 0) {
 		struct protocol_t *protocol = NULL;
-		nrfilter = explode(filter, ",", &filters);
 		unsigned int match = 0, j = 0;
 
 		protocol_init();
 
+		nrfilter = explode(filter, ",", &filters);
```

Comment:
I did in fact check with valgrind and it showed up fine 😕… Thought I had this free'd, apparently I don't, but it still works. Wtf? 😕
Will fix though.